### DISCUSSION TITLE
.NET Aspire Preview 4 has been released

### DISCUSSION THREAD

**Discussion by IEvangelist** [microsoft/aspire#2815] [Announcements]
.NET Aspire Preview 4 is now available! Here's a summary of what's new in this preview release:

- Podman
  - Support for running apps with `podman`
- Dashboard
  - Standalone container image, see [sample](https://learn.microsoft.com/samples/dotnet/aspire-samples/aspire-standalone-dashboard)
  - Accessibility improvements, including shortcuts with <kbd>Shift</kbd>+<kbd>?</kbd>
  - Metrics table view, toggle between graph or table
- Databases and Entity Framework improvements
  - New management tools, MySQL (phpMyAdmin) and MongoDB (mongo-express)
  - Entity Framework migrations issues resolved, see [sample](https://learn.microsoft.com/samples/dotnet/aspire-samples/aspire-efcore-migrations)
  - Database server resource updates
  - Database container resource updates
  - New `Enrich` APIs, `Enrich[Provider]DdContext`
- External parameters, see [.NET Aspire docs: External parameters](https://learn.microsoft.com/en-us/dotnet/aspire/fundamentals/external-parameters)
- New idioms:
  - `DistributedApplicationBuilder.ExecutionContext`, and the `IsPublisherMode`
  - `PublishAs`, `RunAs`, and `As` APIs
- API improvements
  - Change container properties, `WithImageTag`
  - Splitting bind and volume mounts, `WithVolumeMount` and `WithBindMount`
- Components
  - **New**: [Azure AI Search Documents](https://learn.microsoft.com/en-us/dotnet/aspire/azureai/azureai-search-document-component?tabs=dotnet-cli)
  - **Scenario**: [.NET Aspire support for Azure SignalR Service](https://learn.microsoft.com/en-us/dotnet/aspire/real-time/azure-signalr-scenario?tabs=dotnet-cli)
  - **Update** : [.NET Aspire with Application Insights](https://learn.microsoft.com/en-us/dotnet/aspire/deployment/azure/application-insights)
- Containers with Azure resource mapping (with more planned)
  - Redis -> Azure Redis
  - Postgres -> Azure Database for PostgresSQL
  - SQL Server -> Azure SQL Server
- Deployment
  - New primitive, the Azure Bicep resource
  - AZD has been updated to include better support for deployment scenarios to align with `preview-4` of .NET Aspire
  - The deployment manifest now only contains six resource types, see [.NET Aspire manifest format for deployment tool builders](https://learn.microsoft.com/en-us/dotnet/aspire/deployment/manifest-format)
- Emulators updates
  - Change from `UseEmulator` to `RunAsEmulator`

For more information, see [.NET Aspire docs: Preview 4](https://learn.microsoft.com/en-us/dotnet/aspire/whats-new/preview-4).

**Comment by KalleOlaviNiemitalo** [comment]
I don't see a 8.0 preview 4 tag at <https://github.com/dotnet/aspire/tags> yet.

**Reply by IEvangelist** [reply]
Yes, I believe @joperezr still needs to create it.

**Reply by joperezr** [reply]
Hello @KalleOlaviNiemitalo, thanks for raising this. This is expected, as we are just waiting on merging the internal branch that is used to build the releases into GH so that we can get the right SHA. We typically do this 1 or 2 days after the actual release, so hopefully I can get this done today.

**Reply by joperezr** [reply]
Release is out now: https://github.com/dotnet/aspire/releases/tag/v8.0.0-preview.4.24156.9
